app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserMenuSignIn.kt

```diff
@@ -0,0 +1,71 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar
+
+import android.content.Context
+import android.view.View
+import android.widget.TextView
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.ContextCompat
+import mozilla.components.browser.menu.BrowserMenu
+import mozilla.components.browser.menu.item.BrowserMenuImageText
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+
+/**
+ * A menu item for displaying account information. The item calculates the label on every bind call,
+ * to provide each menu with the latest account manager state.
+ *
+ * @param label The function that provides the visible label of this menu item.
+ * @param imageResource ID of a drawable resource to be shown as icon.
+ * @param textColorResource Optional ID of color resource to tint the text.
+ * @param listener Callback to be invoked when this menu item is clicked.
+ */
+
+class BrowserMenuSignIn(
+    @ColorRes private val textColorResource: Int,
+    @DrawableRes imageResource: Int = R.drawable.ic_signed_out,
+    listener: () -> Unit = {},
+) : BrowserMenuImageText(
+    String(),
+    imageResource,
+    textColorResource = textColorResource,
+    listener = listener,
+) {
+    override fun bind(menu: BrowserMenu, view: View) {
+        super.bind(menu, view)
+        val textView = view.findViewById<TextView>(R.id.text)
+        textView.text = getLabel(textView.context)
+        textView.setTextColor(ContextCompat.getColor(textView.context, textColorResource))
+    }
+
+    /**
+     * Determines the proper label for the sign in button.  There are three states:
+     * user is signed-in, please sign-in, and we are not sure what's happening. The last state does
+     * occur when the menu is opened before the AccountManager has finished updating its state, or
+     * when there is no connection and AccountManager doesn't return us the profile because its cache
+     * has 2 minutes 'freshness':
+     * https://searchfox.org/mozilla-mobile/source/application-services/components/fxa-client/src/internal/profile.rs#41
+     */
+    @VisibleForTesting
+    internal fun getLabel(context: Context): String = with(context) {
+        val isSignedIn = components.settings.signedInFxaAccount
+        val email = components.backgroundServices.syncStore.state.account?.email
+
+        if (isSignedIn) {
+            // It sounds a bit counterintuitive, but singed-in state doesn't mean we
+            // have the profile information. The state can be (and usually is) restored
+            // without connection to the Internet, but profile cache is accessible just
+            // for two minutes. So we have to make sure that we actually have the email
+            // in order to show it. Improving cache work progress:
+            // https://github.com/mozilla/application-services/issues/5328
```

Comment:
Could we remove this comment as well? I imagine this is no longer an issue that no longer needs to documented due to our `SyncStore` usage.